scality_sproxyd_client/sproxyd_client.py

```diff
@@ -63,6 +64,10 @@ def __init__(self, sproxyd_urls, conn_timeout=None, proxy_timeout=None,
         :type logger: `logging.Logger`
         '''
 
+        if not sproxyd_urls:
```

Comment:
This doesn't work when the caller passes in a generator of URLs, like
```python
client = SproxydClient(endpoint.url for endpoint in endpoints, conn_timeout=10)
```

Better first convert the given list into a set. Note I reworked this a bit in https://github.com/scality/ScalitySproxydSwift/commit/bb5173ff58eaa99ffd7bdb8b034ff3d3d31a8ee5